app/[locale]/template/page.tsx

```diff
@@ -11,110 +13,268 @@ import {
-      content: (
-        <div>
-          <p>National Science Foundation (nsf.gov)</p>
-          <p>Last updated: 04-01-2024</p>
-        </div>
-      ),
-      funder: 'National Science Foundation',
-      lastUpdated: '04-01-2024',
-      publishStatus: 'Published',
-      defaultExpanded: false,
-    },
-    {
-      title: 'NSF Polar Expeditions',
-      link: '/template/tpl_Ad525',
-      content: (
-        <div>
-          <p>National Science Foundation (nsf.gov)</p>
-          <p>Last updated: 06-15-2024</p>
-          <p>Status: Unpublished</p>
-        </div>
-      ),
-      funder: 'National Science Foundation',
-      lastUpdated: '06-15-2024',
-      publishStatus: 'Unpublished',
-      defaultExpanded: false,
-    },
-    {
-      title: 'NSF: McMurdo Station (Antarctic)',
-      link: '/template/tpl_Ad525',
-      content: (
-        <div>
-          <p>National Science Foundation (nsf.gov)</p>
-          <p>Last updated: 09-21-2024</p>
-        </div>
-      ),
-      funder: 'National Science Foundation',
-      lastUpdated: '09-21-2024',
-      publishStatus: 'Published',
-      defaultExpanded: false,
+  // Make graphql request for templates under the user's affiliation
+  const { data = {}, loading, error: queryError, refetch } = useTemplatesQuery({
+    /* Force Apollo to notify React of changes. This was needed for when refetch is
+    called and a re-render of data is necessary*/
+    notifyOnNetworkStatusChange: true,
+  });
+
+  //Update searchTerm state whenever entry in the search field changes
+  const handleSearchInput = (value: string) => {
+    setSearchTerm(value);
+  }
+
+  // Filter results when a user enters a search term and clicks "Search" button
+  const handleFiltering = (term: string) => {
+    setErrors([]);
+    const filteredList = templates.filter(item => item.title.includes(term));
+    if (filteredList.length >= 1) {
+      setSearchTerm(term);
+      setFilteredTemplates(filteredList);
+    } else {
+      //If there are no matching results, then display an error
+      const errorMessage = t('noItemsFoundError', { term })
+      setErrors(prev => [...prev, errorMessage]);
+    }
+  }
+
+  const fetchTemplateVersions = async (templateId: number) => {
+    try {
+      const { data } = await client.query<TemplateVersionsQuery, TemplateVersionsQueryVariables>({
+        query: TemplateVersionsDocument,
+        variables: { templateId },
+      });
+      return data;
+    } catch (err) {
+      if (err instanceof ApolloError) {
+        const { data } = await client.query<TemplateVersionsQuery, TemplateVersionsQueryVariables>({
+          query: TemplateVersionsDocument,
+          variables: { templateId },
+        });
+        return data;
+      } else {
+        console.error('Error fetching template versions:', err);
+      }
     }
-  ];
+  }
+
+  // Get all template versions in order to get latest one with its versionType
+  const fetchLatestVersion = async (templateId: number) => {
```

Comment:
Oh ok, thanks @briri. I totally forgot that `isDirty` indicates "PUBLISHED". That will definitely simplify the code as well. I'll make that change.

Reply:
- `currentVersion` blank (`isDirty` should always be true here) => never been published
- `currentVersion` not blank and `isDirty` is false => this is the current published version
- `currentVersion` not blank and `isDirty` is true => the template has been modified since publication and the changes need to be published

By the way, you do not need to manage either of these properties on the UI side. They are updated by the backend as changes are made to the record

Reply:
Thanks @briri. I remove the versionedTemplates query, and am using the `isDirty` boolean to determine whether the template was published or not. Thanks again.